libipl/p10/ipl0.C

```diff
@@ -242,10 +242,43 @@ static int update_hwas_state_callback(struct pdbg_target* target, void *priv)
 	return GUARD_TGT_FOUND;
 }
 
-//@Brief Function will get the guard records and will unset the functional
-//state of the guarded resources in HWAS state attribute in device tree.
-static void update_hwas_state(bool is_coldboot)
+/**
+ * @brief Allow guard actions in the below cases to support resource recovery.
+ *
+ * - If the current boot is MPIPL.
+ *
+ * - If the BOOTTIME_GUARD_INDICATOR file (that will be created by the BMC
+ *   in the PowerOn or TI or Checkstop or Watchdog timeout path) is exist
+ *   in the current boot.
+ */
+static bool guard_action_allowed()
 {
+    namespace fs = std::filesystem;
+    constexpr auto BOOTTIME_GUARD_INDICATOR = "/tmp/phal/boottime_guard_indicator";
+    fs::path boottime_guard_indicator(BOOTTIME_GUARD_INDICATOR);
+
+    if (ipl_type() == IPL_TYPE_MPIPL) {
```

Comment:
Better to move this chcek at begining. 256-258 line is not really consumed in thempipl path.

Reply:
Ack